discovery/reliable_sender.go

```diff
@@ -225,6 +225,14 @@ out:
 		// can only contain messages which have a ShortChannelID field.
 		shortChanID, _ := msgShortChanID(msg)
 
+		// Ensure the peer is still online right before sending the
+		// message.
+		select {
+		case <-offlineChan:
```

Comment:
Why's this needed, if we go to send the message below, we'll get an error and go back into `waitUntilOnline` anyway. Or is there a possibility of deadlock?

Reply:
The send below is asynchronous, which I believe won't return an error back if the peer disconnects.

Reply:
Could alternatively make the send synchronous instead.